pkg/operation/botanist/migration.go

```diff
@@ -0,0 +1,168 @@
+	fns, err = b.applyFuncToAllExtensionCRs(ctx, annotateObjectForMigrationFunc(ctx, b.K8sSeedClient.Client()))
+	if err != nil {
+		return err
+	}
+
+	return flow.Parallel(fns...)(ctx)
+}
+
+func annotateObjectForMigrationFunc(ctx context.Context, client client.Client) func(runtime.Object) error {
+	return func(obj runtime.Object) error {
+		return kutil.TryUpdate(ctx, k8sretry.DefaultBackoff, client, obj, func() error {
+			acc, err := extensions.Accessor(obj)
+			if err != nil {
+				return err
+			}
+
+			kutil.SetMetaDataAnnotation(acc, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationMigrate)
+			return nil
+		})
+	}
+}
+
+// DeleteAllExtensionCRs deletes all extension CR from the Shoot namespace
+func (b *Botanist) DeleteAllExtensionCRs(ctx context.Context) error {
+	fns, err := b.applyFuncToAllExtensionCRs(ctx, func(obj runtime.Object) error {
+		if err := common.ConfirmDeletion(ctx, b.K8sSeedClient.Client(), obj); err != nil {
+			return err
+		}
+
+		if err := client.IgnoreNotFound(b.K8sSeedClient.Client().Delete(ctx, obj, kubernetes.DefaultDeleteOptions...)); err != nil {
+			accObj, err := meta.Accessor(obj)
+			if err != nil {
+				return err
+			}
+			return fmt.Errorf("couldn't delete CR %s with name %s: %v", obj.GetObjectKind(), accObj.GetName(), err)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return flow.Parallel(fns...)(ctx)
+}
+
+// WaitForExtensionsOperationMigrateToSucceed waits until extension CR has lastOperation Migrate Succeeded
+func (b *Botanist) WaitForExtensionsOperationMigrateToSucceed(ctx context.Context) error {
+	fns, err := b.applyFuncToAllExtensionCRs(ctx, func(obj runtime.Object) error {
+		return retry.UntilTimeout(ctx, 5*time.Second, 300*time.Second, func(ctx context.Context) (done bool, err error) {
+			acc, err := extensions.Accessor(obj)
+			if err != nil {
+				return retry.SevereError(err)
+			}
+
+			exrtensionObjStatus := acc.GetExtensionStatus()
+			if exrtensionObjStatus != nil && exrtensionObjStatus.GetLastOperation() != nil {
+				lastOperation := exrtensionObjStatus.GetLastOperation()
+				if lastOperation != nil && lastOperation.GetType() == gardencorev1beta1.LastOperationTypeMigrate && lastOperation.GetState() == gardencorev1beta1.LastOperationStateSucceeded {
+					return retry.Ok()
+				}
+			}
+
+			var exetnsionType string
+			if extensionSpec := acc.GetExtensionSpec(); extensionSpec != nil {
+				exetnsionType = extensionSpec.GetExtensionType()
+			}
+			return retry.MinorError(fmt.Errorf("extension CR %s with type %s lastOperation was not Migrate=Succeeded", acc.GetName(), exetnsionType))
+		})
+	})
+	if err != nil {
+		return err
+	}
+
+	return flow.Parallel(fns...)(ctx)
+}
+
+func (b *Botanist) applyFuncToAllExtensionCRs(ctx context.Context, applyFunc func(obj runtime.Object) error) ([]flow.TaskFn, error) {
+	var fns []flow.TaskFn
+	for _, listObj := range extensions.GetAllGardenerExtensionsLists() {
+		listObjCopy := listObj.DeepCopyObject()
```

Comment:
`GetAllGardenerExtensionsLists()` will generate new lists so there's no need for `DeepCopyObject()`